app/controllers/users_controller.rb

```diff
@@ -14,57 +16,41 @@ def create
   end
 
   def show
-    @user = User.find(params[:id])
     @friendship = Friendship.find_by(user_id: @user.id, friend_id: current_user.id) || Friendship.find_by(user_id: current_user.id, friend_id: @user.id)
     if current_user.id == @user.id
       user_path(@user)
-    elsif current_user.friends.include?(@user) && @friendship.approved
-      user_path(@user)
-    elsif current_user.inverse_friends.include?(@user) && @friendship.approved
+    elsif current_user.all_friends.include?(@user) && @friendship.approved
       user_path(@user)
     else
       render "private"
     end
   end
 
-  def autocomplete_username
-    term = params[:q].downcase
-    @users = User.where("lower(username) LIKE '%#{term}%'")
-    respond_to do |format|
-      format.json {
-        render json: @users.map(&:username)
-      }
-    end
```

Comment:
@c0d3s Ah ok. I think @djvoa12's last minute changes to include tabs might've fixed the problem. At the time I was working on it, it was all on the same page. 

I'm slowly working on being able to test locally, but putting off dealing with weird postgres issues for now. I'll get to it this weekend.